src/loader/loader.c

```diff
@@ -215,6 +253,14 @@ post_analyze_hook(ParseState *pstate, Query *query)
 	}
 }
 
+static void
+timescale_shmem_startup_hook(void)
+{
+	if (prev_shmem_startup_hook)
+		prev_shmem_startup_hook();
+	timescale_bgw_shmem_startup();
```

Comment:
two things: 1) naming consistency between timescale_bgw and tsbgw.  2) the bgw_shmem startup hook should call the startup hooks of all bgw components (the queue, worker counter, etc). The loader.c should know about "bgw" not "message queue for bgw"

Reply:
also goes for shmem_init below